### PR TITLE
Upgrade to React 16.6.3

### PR DESCRIPTION
**Pull request by alecmerdler**
### Description

Updates our view render framework and associated types to let us use new lifecycle hooks.

### CODE REVIEW

File: frontend/public/components/factory/list.tsx [hunk 1]
```diff
@@ -402,10 +408,18 @@ export const List = connect(stateToProps, {sortList: UIActions.sortList})(
       selector: PropTypes.object,
       staticFilters: PropTypes.array,
       virtualize: PropTypes.bool,
+      currentSortField: PropTypes.string,
```
Comment (alecmerdler) [inline frontend/public/components/factory/list.tsx]: Had to adjust the `PropTypes` to match the TypeScript props `type`, which is pretty neat that it checks it now.

**Comment by spadgett** [comment]
lgtm, @alecmerdler can you rebase to pick up dependency changes from #713? @sosiouxme will need that to update yarn offline for downstream builds.

**Comment by sosiouxme** [comment]
@alecmerdler please include the following change to the `sources` file:

```
--- a/sources
+++ b/sources
@@ -1,2 +1,2 @@
 401aa1438604b8e32306fa2bd4f8e211  node-v8.9.4-headers.tar.gz
-5a31a2c68718575bd3b1d239e9b2c103  yarn-offline.tar
+8ea694cca7a41d22979be0d8a4846bd1  yarn-offline.tar
```

**Pull request merged**